src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/elasticapi/ElasticExtensions.kt

```diff
@@ -98,3 +106,26 @@ fun IndexMetadata.getManagedIndexMetaData(): ManagedIndexMetaData? {
     }
     return null
 }
+
+/**
+ * Do a exists search query to retrieve all policy with ism_template field
+ * parse search response with this function
+ *
+ * @return map of policyID to ISMTemplate in this policy
+ */
+fun getPolicyToTemplateMap(response: SearchResponse, xContentRegistry: NamedXContentRegistry = NamedXContentRegistry.EMPTY):
+    Map<String, ISMTemplate?> {
+    return response.hits.hits.map {
+        val id = it.id
+        val seqNo = it.seqNo
+        val primaryTerm = it.primaryTerm
+        val xcp = XContentFactory.xContent(XContentType.JSON)
+            .createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, it.sourceAsString)
+        xcp.parseWithType(id, seqNo, primaryTerm, Policy.Companion::parse)
```

Comment:
This can throw an exception right?

Reply:
used in:
1. Coordinator: catch in the general exception, logger out the error, return empty
2. TransportIndexPolicyAction: directly throw `IllegalArgumentException`